#VRML_SIM V7.3.0 utf8

PROTO CozmoVizDisplay 
[ 
  # Resolution of incoming camera images.
  # Set this to match IMG_STREAM_RES in cozmoConfig.h
  field SFString streamResolution "QVGA"
  
  # Following (but with PixelSize=2) works well on Macbook screen (size=1 for monitor)
  field SFFloat pixelSize 1.0

  field SFBool moodViz FALSE
  field SFVec2f moodPos 1.0 0.25
  field SFBool behaviorSelectionViz FALSE
  field SFVec2f behaviorSelectionPos 1.0 0.50

  field SFBool dockViz FALSE
  field SFBool infoViz TRUE
  field SFBool camViz TRUE
]
{
Supervisor {
  
  children [
    DEF DOCKING_DISPLAY Display {
      # Hide display plane far away
      translation 1000 0.1 0

      children [
        DEF SHAPE Shape {
          appearance Appearance {
            material Material {
              diffuseColor 1 1 1
              specularColor 0 0 0
              transparency 0.6
            }
          }
          geometry Plane {
	          size 1 1
          }
        }
      ]
      name "cozmo_docking_display"
      width 200
      height 240
      windowPosition 1 1
      %{ if fields.dockViz.value then }%
        pixelSize %{=fields.pixelSize.value}%
      %{ else }%
        pixelSize 0
      %{ end }%
    }


    DEF DISPLAY_PLANE Display {
      # Hide display plane far away
      translation 1000 0.1 0

      children [
        DEF SHAPE Shape {
          appearance Appearance {
            material Material {
              diffuseColor 1 1 1
              specularColor 0 0 0
              transparency 0.6
            }
          }
          geometry Plane {
	          size 1 1
          }
        }
      ]
      name "cozmo_viz_display"
      width 400
      height 200
      windowPosition 1 1
      %{ if fields.infoViz.value then }%
        pixelSize %{=fields.pixelSize.value}%
      %{ else }%
        pixelSize 0
      %{ end }%
    }
    
    DEF DISPLAY_PLANE Display {
      # Hide display plane far away
      translation 1000 0.1 0

      children [
        DEF SHAPE Shape {
          appearance Appearance {
            material Material {
              diffuseColor 1 1 1
              specularColor 0 0 0
              transparency 0.6
            }
          }
          geometry Plane {
            size 1 1
          }
        }
      ]
      name "cozmo_mood_display"
      width 450
      height 150
      windowPosition %{=fields.moodPos.value.x}% %{=fields.moodPos.value.y}%
      %{ if fields.moodViz.value then }%
        pixelSize %{=fields.pixelSize.value}%
      %{ else }%
        pixelSize 0
      %{ end }%
    }
    
    DEF DISPLAY_PLANE Display {
      # Hide display plane far away
      translation 1000 0.1 0

      children [
        DEF SHAPE Shape {
          appearance Appearance {
            material Material {
              diffuseColor 1 1 1
              specularColor 0 0 0
              transparency 0.6
            }
          }
          geometry Plane {
            size 1 1
          }
        }
      ]
      name "cozmo_behavior_display"
      width 450
      height 150
      windowPosition %{=fields.behaviorSelectionPos.value.x}% %{=fields.behaviorSelectionPos.value.y}%
      %{ if fields.behaviorSelectionViz.value then }%
        pixelSize %{=fields.pixelSize.value}%
      %{ else }%
        pixelSize 0
      %{ end }%
    }

    DEF CAM_DISPLAY_PLANE Display {
      # Hide display plane far away
      translation 1000 0.1 0

      children [
        DEF SHAPE Shape {
          appearance Appearance {
            material Material {
              diffuseColor 1 1 1
              specularColor 0 0 0
              transparency 0.6
            }
          }
          geometry Plane {
	         size 1 1
          }
        }
      ]
      name "cozmo_cam_viz_display"


      %{ if fields.streamResolution.value == "VGA" then }%
      width 640
      height 480
      #pixelSize 0.5
      %{ elseif fields.streamResolution.value == "QVGA" then }%
      width 320
      height 240
      #pixelSize 1
      %{ elseif fields.streamResolution.value == "QQVGA" then }%
      width 160
      height 120
      #pixelSize 2
      %{ elseif fields.streamResolution.value == "QQQVGA" then }%
      width 80
      height 60
      #pixelSize 4
      %{ elseif fields.streamResolution.value == "CVGA" then }%
      width 400
      height 296
      #pixelSize 1
      %{ end }%

      %{ if fields.camViz.value then }%
        pixelSize %{=fields.pixelSize.value}%
      %{ else }%
        pixelSize 0
      %{ end }%

      windowPosition 0 1
    }
  ]

  controller "webotsCtrlViz"
} # Supervisor

} # PROTO